src/Workspaces/Core/Portable/Utilities/TaskExtensions.cs

```diff
@@ -60,8 +60,10 @@ public static T WaitAndGetResult<T>(this Task<T> task, CancellationToken cancell
         // thread.  In the future we are going ot be removing this and disallowing its use.
         public static T WaitAndGetResult_CanCallOnBackground<T>(this Task<T> task, CancellationToken cancellationToken)
         {
-            task.Wait(cancellationToken);
-            return task.Result;
+            // WaitAny waits for the task to complete with cancellation and without throwing;
+            // GetAwaiter().GetResult() is then used so that if the task failed, the exception is not wrapped in AggregateException
+            Task.WaitAny(new Task[] { task }, cancellationToken);
```

Comment:
My guess is that blocking on async code (which is what this method seems to be used for) should be fairly rare, so a single allocation should be fine. And I would expect that calling it with a completed task should be rare.

But I don't actually know this for certain.